spark-job-server-proposal.md

```diff
@@ -0,0 +1,75 @@
+# Adding Spark Job Server To Run Spark Jobs
+
+## Introduction
+
+This document provides design of spark job server which would trigger spark
+jobs either as scheduled or batch jobs. Providing the user a ui for inputing
+the source code git location, environment variables required to integrate with
+other apps, input parameters for spark job, selecting s2i builder image and
+files required.
+
+## Problem Statement
+
+The Data Platform Spark applications can only run within OpenShift
+environment and users are required to do provide only source code which
+currently we use source to image to create a standard image which will be run
+against the spark cluster. Currently there is no way to schedule jobs or to run
+cron jobs. Also if users have a more complex setup where they would need
+JDG or other servers they would be required to do a lot of manual setup.
+
+## Proposed Solution
+
+Apache Spark users will submit their spark job in the form of binary jar or py
+files. These files will require users to provide input as well as be triggered
+by spark-submit shell script. However there are other environment requirements
+that may be needed by users as well.
+
+The requirements to create, run, rerun, stop a particular spark job by one
+click of a button.
+
+With a simple html form users can provide the following input:
+* source git url
+* input parameters for spark job
+* files required to be mounted to run the job.
+* inject environment variables from services that this spark job would need
+to integrate with.
+* specify if they would like a new cluster created or if they would like to use
+an existing one
+* schedule when the job should run
+* how often the job should run
+
+The workflow would be user fills out html form and oshinko-console would send
+json payload with the information required to build the s2i image and then run
+a spark job including the inputs provided above. After that we would get an
+update in the ui to see jobs running and their status. There would be a button
+on the ui for users to rerun the spark job.
+
```

Comment:
i would like to see more discussion of what will be happening behind the scenes with respect to the "spark job" and how it will be run and monitored through kubernetes.

for example, if we plan to use kubernetes Job objects, these are meant to be used in cases where the pod will run to a successful completion, and on failure kubernetes will attempt to restart the pod. i'd like to know more details about how this will be implemented in kubernetes.